### PR TITLE
streamlink-twitch-gui-bin: fix missing player error

### LINKED ISSUE #133488: `streamlink-twitch-gui-bin`: missing runtime dependencies
### Describe the bug
Cannot view any content when running the `strealink-twitch-gui-bin` program because it complains about missing binaries.

### Steps To Reproduce
1. run `nix-shell -I nixpkgs=/nix/var/nix/profiles/per-user/$USER/channels/nixos-unstable -p streamlink-twitch-gui-bin`
2. run `streamlink-twitch-gui`
3. try to watch any stream

### Expected behavior
No errors, stream works

### Notify maintainers
@rileyinman 

### Metadata
`[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"`
 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.108, NixOS, 20.09.3670.310885ec804 (Nightingale)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(admin): `"nixos-unstable-21.11pre304626.8ecc61c91a5, nixpkgs-20.09.4393.068984c00e0"`
 - nixpkgs: `/home/$USER/.nix-defexpr/channels/nixpkgs`

### Possible fix

Add `VLC` and `streamlink` as runtime dependencies because the following works for me:
`nix-shell -I nixpkgs=/nix/var/nix/profiles/per-user/$USER/channels/nixos-unstable -p streamlink-twitch-gui-bin streamlink vlc`

### PR DESCRIPTION
**Pull request by emilytrau**
###### Motivation for this change

Fixes #133488

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Artturin** [comment]
`Error: error: The default player (VLC) does not seem to be installed`

could you apply this patch?

```patch
diff --git a/pkgs/applications/video/streamlink-twitch-gui/bin.nix b/pkgs/applications/video/streamlink-twitch-gui/bin.nix
index 580408d36d3..666e3f66f6e 100644
--- a/pkgs/applications/video/streamlink-twitch-gui/bin.nix
+++ b/pkgs/applications/video/streamlink-twitch-gui/bin.nix
@@ -24,11 +24,12 @@
 , nspr
 , xorg
 , streamlink
+, vlc
 }:
 let
   basename = "streamlink-twitch-gui";
   runtimeLibs = lib.makeLibraryPath [ libudev0-shim ];
-  runtimeBins = lib.makeBinPath [ streamlink ];
+  runtimeBins = lib.makeBinPath [ streamlink vlc ];
   arch =
     if stdenv.hostPlatform.system == "x86_64-linux"
     then
@@ -96,11 +97,16 @@ stdenv.mkDerivation rec {
     # Install all files, remove unnecessary ones
     cp -a . $out/opt/${basename}/
     rm -r $out/opt/${basename}/{{add,remove}-menuitem.sh,credits.html,icons/}
-
-    wrapProgram $out/opt/${basename}/${basename} --add-flags "--no-version-check" --prefix LD_LIBRARY_PATH : ${runtimeLibs} --prefix PATH : ${runtimeBins}
-
     ln -s "$out/opt/${basename}/${basename}" $out/bin/
-    ln -s "${desktopItem}/share/applications" $out/share/
+    cp -r "${desktopItem}/share/applications" $out/share/
+  '';
+
+  preFixup = ''
+    gappsWrapperArgs+=(
+      --add-flags "--no-version-check" \
+      --prefix LD_LIBRARY_PATH : ${runtimeLibs} \
+      --prefix PATH : ${runtimeBins}
+    )
   '';
 
   desktopItem = makeDesktopItem {
```

if someone could submit a pr upstream to add `/run/current-system/sw/bin` to here https://github.com/streamlink/streamlink-twitch-gui/blob/9be126dc135f891809f6033944ef955ad2bcd4c8/src/config/players.json#L19
then we wouldn't need to add it to path ( i think )

**Comment by angustrau** [comment]
@bastimeyer thanks for dropping by! We could add vlc to the wrapper for streamlink instead. Is vlc a hard dependency of either streamlink or streamlink-gui?

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 133568` at 5d4534d4 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>streamlink-twitch-gui-bin</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-uri-literals.md">no-uri-literals</a>

URI literals are deprecated.
Near pkgs/applications/video/streamlink-twitch-gui/bin.nix:125:20:
```
    |
125 |     downloadPage = https://github.com/streamlink/streamlink-twitch-gui/releases;
    |                    ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/video/streamlink-twitch-gui/bin.nix:94:3:
```
   |
94 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Comment by bastimeyer** [comment]
> Is vlc a hard dependency of either streamlink or streamlink-gui?

As said, neither have it as a hard dependency. Both let you choose whichever player you want, and Streamlink defaults to `vlc` (read from `PATH`) if no `--player` has been set via its CLI or via its config file(s). Streamlink Twitch GUI defaults to the default behavior of Streamlink, so it doesn't have any player dependencies at all and you shouldn't set one here.

**Review by SuperSandro2000 — APPROVED** [review]
didn't test but the diff LGTM

File: pkgs/applications/video/streamlink-twitch-gui/bin.nix [hunk 1]
```diff
@@ -28,6 +28,7 @@
 let
   basename = "streamlink-twitch-gui";
   runtimeLibs = lib.makeLibraryPath [ libudev0-shim ];
+  runtimeBins = lib.makeBinPath [ streamlink ];
```
Comment (Donaim) [inline pkgs/applications/video/streamlink-twitch-gui/bin.nix:31]: What about "VLC"? I think it is used by streamlink to play the streams.

**Pull request merged**